tests/v2/integration/steveapi/steve_api_test.go

```diff
@@ -0,0 +1,1543 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	projectNamePrefix = "test-project"
+	namespacePrefix   = "test-ns"
+	labelKey          = "test-label"
+	continueToken     = "nondeterministictoken"
+	revisionNum       = "nondeterministicint"
+)
+
+var (
+	userEnabled                = true
+	continueReg                = regexp.MustCompile(`(continue=)[\w=]+(%3D){0,2}`)
+	namespaceSecretManagerRole = rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "namespace-secret-manager",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs: []string{
+					"get",
+					"list",
+				},
+				APIGroups: []string{
+					"",
+				},
+				Resources: []string{
+					"secrets",
+				},
+			},
+		},
+	}
+	mixedSecretUserRole = rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "mixed-secret-user",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs: []string{
+					"get",
+					"list",
+				},
+				APIGroups: []string{
+					"",
+				},
+				Resources: []string{
+					"secrets",
+				},
+				ResourceNames: []string{
+					"test1",
+					"test2",
+				},
+			},
+		},
+	}
+	testUsers = map[string]interface{}{
+		"user-a": management.ProjectRoleTemplateBinding{
+			RoleTemplateID: "project-owner",
+		},
+		"user-b": []rbacv1.RoleBinding{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "namespace-secret-manager",
+					Namespace: "test-ns-1",
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: rbacv1.SchemeGroupVersion.Group,
+					Kind:     "Role",
+					Name:     "namespace-secret-manager",
+				},
+			},
+		},
+		"user-c": []rbacv1.RoleBinding{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mixed-secret-user",
+					Namespace: "test-ns-1",
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: rbacv1.SchemeGroupVersion.Group,
+					Kind:     "Role",
+					Name:     "mixed-secret-user",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mixed-secret-user",
+					Namespace: "test-ns-2",
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: rbacv1.SchemeGroupVersion.Group,
+					Kind:     "Role",
+					Name:     "mixed-secret-user",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mixed-secret-user",
+					Namespace: "test-ns-3",
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: rbacv1.SchemeGroupVersion.Group,
+					Kind:     "Role",
+					Name:     "mixed-secret-user",
+				},
+			},
+		},
+	}
+	namespaceMap = map[string]string{
+		"test-ns-1": "",
+		"test-ns-2": "",
+		"test-ns-3": "",
+		"test-ns-4": "",
+		"test-ns-5": "",
+	}
+)
+
+type SteveAPITestSuite struct {
+	suite.Suite
+	client            *rancher.Client
+	session           *session.Session
+	project           *management.Project
+	userClients       map[string]*rancher.Client
+	lastContinueToken string
+	lastRevision      string
+}
+
+func (s *SteveAPITestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *SteveAPITestSuite) SetupSuite() {
+	testSession := session.NewSession(s.T())
```

Comment:
This goes for the namespace creation, secrets, and role creation beforehand. The reason you are using the kubeapi extensions is you want to create these resources with the controllers, and then verify behavior with the Steve client? I'm just curious, and a little description will help understand.

Reply:
I want to avoid using the norman API as much as possible since we are phasing it out. I could have created these through steve, but since I don't care about testing how these resources are created, the simplest way to create a resource in kubernetes is to use kubernetes itself, so that was my preference.

Reply:
Sounds good, just was curious. Thank you!